test/unit/services/awsAccess.test.ts

```diff
@@ -0,0 +1,127 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import {
+  IdentitystoreClient,
+  ListUsersCommand,
+  CreateUserCommand,
+  CreateGroupMembershipCommand,
+} from "@aws-sdk/client-identitystore";
+import {
+  createUser,
+  addUserToGroup,
+  fetchAwsUserIdByUsername,
+} from '../../../utils/awsFunctions';
+
+describe('AWS Identity Store Functions', () => {
+  let sendStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    // Reset all stubs before each test
+    sinon.restore();
+    
+    // Create a stub for the send method
+    sendStub = sinon.stub(IdentitystoreClient.prototype, 'send');
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('fetchAwsUserIdByUsername', () => {
+    it('should return userId when user is found', async () => {
+      const mockResponse = {
```

Comment:
can we rename this to expectedResponse.

Reply:
since we are not asserting it with any other response, I named it like mock response